### PR TITLE
Add controller-gen to tools.go

### PR DESCRIPTION
**Pull request by mmlb**
## Description

Modifies most of kube.mk to be better integrated with tools.go and accompanying make rules.

## Why is this needed

Gets rid of recursive make calls (which are considered harmful :smile:) by giving make more/better info.
Also removes code that is no longer necessary (and some that never was, envsubst in kube.mk).

## How Has This Been Tested?

I ran `make generate`

### CODE REVIEW

File: kube.mk [hunk 1]
```diff
@@ -1,36 +1,18 @@
...
-	$(CONTROLLER_GEN) \
+generate-manifests: bin/controller-gen # Generate manifests e.g. CRD, RBAC etc.
+	controller-gen \
 		paths=./pkg/apis/... \
 		crd:crdVersions=v1 \
 		rbac:roleName=manager-role \
 		output:crd:dir=./config/crd/bases \
 		output:webhook:dir=./config/webhook \
 		webhook
+	prettier --write ./config/crd/
```
Comment (micahhausler) [inline kube.mk:18]: 👍

**Review by micahhausler — APPROVED** [review]
LGTM, thanks for the cleanup. I mostly just copied what kubebuilder dumps out and what was in CAPT

File: kube.mk (hunk 1 above)
Comment (mmlb) [reply kube.mk:18]: :nod: but now I have to fix ci.

Comment (mmlb) [reply kube.mk:18]: should be fixed now

**Review by micahhausler — APPROVED** [review]
LGTM

**Review by micahhausler — APPROVED** [review]
LGTM

**Review by ScottGarman — APPROVED** [review]

**Pull request merged**